src/views/Login/PersonaRegistration/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useKeycloak } from '@react-keycloak/web';
+import { KidsFirstKeycloakTokenParsed } from 'common/tokenTypes';
+import { fetchPersonaUser } from 'store/persona/thunks';
+import { REDIRECT_URI_KEY } from 'common/constants';
+import useQueryParams from 'hooks/useQueryParams';
+import { STATIC_ROUTES } from 'utils/routes';
+
+import Registration from './RegistrationForm';
+import TermsConditions from './TermsConditions';
+import { usePersona } from 'store/persona';
+import Spinner from 'components/uiKit/Spinner';
+
+enum Step {
+  LOADING,
+  TERMSANDCONDITIONS,
+  REGISTRATION,
+}
+
+const PersonaRegistration = () => {
+  const { personaUserInfo } = usePersona();
+  const query = useQueryParams();
+  const dispatch = useDispatch();
+  const [step, setStep] = useState(Step.TERMSANDCONDITIONS);
+  const { keycloak } = useKeycloak();
+  const tokenParsed = keycloak?.tokenParsed as KidsFirstKeycloakTokenParsed;
+
+  useEffect(() => {
+    // if the use tried to acces the registration page wihout using keycloack login
+    if (!tokenParsed?.sub!) {
+      window.location.assign(`${window.location.origin}/${STATIC_ROUTES.LOGIN}`);
+      return;
+    }
+
+    if (personaUserInfo === undefined) {
```

Comment:
!personaUserInfo

Reply:
Seems my comment about why I prefer undefined in this case